<?xml version="1.0"?>
<launch>
  <arg name="robot_name" default="robot1"/>
  <arg name="x_pos" default="0.0"/>
  <arg name="y_pos" default="0.0"/>
  <arg name="z_pos" default="0.0"/>

  <!-- Launch Gazebo -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find jackal_gazebo)/worlds/jackal_race.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <!-- Load robot description -->
  <param name="robot_description" 
         command="$(find xacro)/xacro --inorder '$(find cs_bot)/urdf/cs_bot.urdf.xacro'"/>

  <!-- Spawn robot -->
  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" 
        args="-urdf 
              -model $(arg robot_name) 
              -x $(arg x_pos) 
              -y $(arg y_pos) 
              -z $(arg z_pos) 
              -param robot_description" 
        output="screen"/>

  <!-- Publish TF -->
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="50.0"/>
  </node>

</launch>
